src/modules/logger/log_writer_file.cpp

```diff
@@ -644,6 +646,22 @@ bool LogWriterFile::LogFileBuffer::start_log(const char *filename)
 	}
 
 	if (_buffer == nullptr) {
+#if defined(__PX4_NUTTX)
+		struct mallinfo alloc_info = mallinfo();
+
+		// reduced to largest available free chunk, but leave at least 1 kB available
+		static constexpr int one_kb = 1024;
+		const int reduced_buffer_size = (alloc_info.mxordblk - one_kb) / one_kb * one_kb;
+
+		if ((reduced_buffer_size > 0) && ((int)_buffer_size > reduced_buffer_size)) {
```

Comment:
Can you pass the minimum sizes from above (`_min_write_chunk + 300`) and check for these?

Reply:
Updated.